omnipaxos/src/storage.rs

```diff
@@ -728,17 +727,23 @@ where
             }
             self.state_cache.accepted_idx = sync.sync_idx + sync.suffix.len();
             sync_txn.push(StorageOp::AppendOnPrefix(sync.sync_idx, sync.suffix));
-            self.state_cache.stopsign = sync.stopsign.clone();
-            if sync.stopsign.is_some() {
-                self.state_cache.accepted_idx += 1;
+            match sync.stopsign {
+                Some(ss) => {
+                    self.state_cache.stopsign = Some(ss.clone());
+                    self.state_cache.accepted_idx += 1;
+                    sync_txn.push(StorageOp::SetStopsign(Some(ss)));
+                }
+                None => {
```

Comment:
I think if the stopsign is None already, we can probably skip the setting it again in the storage.

```suggestion
                None if self.state_cache.stopsign.is_some() => {
```